lib/identity_cache/query_api.rb

```diff
@@ -193,6 +198,11 @@ def add_cached_associations_to_coder(record, coder)
             hash[name] = record.instance_variable_get(options[:ids_variable_name]) unless options[:embed] == true
           end
         end
+        if (cached_has_ones = klass.cached_has_ones).any? { |name, options| options[:embed] != true }
```

Comment:
Is this an optimization assuming that `options[:embed] == :ids` will be uncommon? I understand we're trying to avoid allocations, but the combination of this line and the `unless options[:embed] == true` 2 lines below is jarring. I don't suppose there's a convenient way to filter once?

Reply:
Without this I think we have to bump the IDC cache version because it changes the shape of all coders / serialized payloads by inserting a `association_id` key and a blank hash value.

Reply:
If anything, it would help avoid bloating the cache blob, which might matter for our large shop blob.

Reply:
I *think* the shop blob is one of the first candidates for the use of this mechanism? I agree with avoiding the IDC cache version bump and blob bloat, if possible.

More concretely, though, my comment relates to how the code reads rather than its effect on serialized payload shape. I wonder whether we can filter once, on the line above, and not need to also filter on line 203 below.

Reply:
Changed to a `select` and `present?` check, which only iterates the hash once.